src/components/input/FilterDropdown.vue

```diff
@@ -0,0 +1,172 @@
+            <div class="pa-2">
+                <v-text-field
+                    v-model="searchText"
+                    label="Search items..."
+                    density="compact"
+                    hide-details
+                    clearable
+                    prepend-inner-icon="mdi-magnify"
+                    variant="outlined"
+                    autofocus
+                ></v-text-field>
+            </div>
+            <v-divider></v-divider>
+            <template v-if="filteredItems.length > 0">
+                <v-list-item v-for="item in filteredItems" :key="item.id">
+                    <v-checkbox v-model="model" :value="item.id" density="compact" hide-details>
+                        <template #label>
+                            <slot :item="item">
+                                {{ item.name }}
+                            </slot>
+                        </template>
+                    </v-checkbox>
+                    <v-tooltip v-if="item.description" :text="item.description" activator="parent" />
+                </v-list-item>
+            </template>
+            <div v-else-if="filteredItems.length === 0" class="pa-4 text-center text-grey">No items found.</div>
+        </v-list>
+    </v-menu>
+</template>
+<script setup lang="ts" generic="T, S extends string, I extends IdObject & { name: string; description?: string }">
+import { computed, PropType, ref, watch } from "vue";
+import { ItemManager } from "@/util/itemManager";
+import { IdObject } from "@/util/types";
+
+const props = defineProps({
+    label: {
+        type: String,
+        required: false,
+        default: "Items"
+    },
+    itemManager: {
+        type: Object as PropType<ItemManager<T, S | "ID">>,
+        required: true
+    },
+    mapper: {
+        type: Function as PropType<(item: T) => I | I[] | undefined | null>,
+        required: true
+    },
+    sorter: {
+        type: Function as PropType<(a: I, b: I) => number>,
+        required: false,
+        default: (a: I, b: I) => a.name.localeCompare(b.name)
+    },
+    additionalInitialValues: {
+        type: Function as PropType<() => Promise<I[]>>,
+        required: false,
+        default: () => Promise.resolve([])
+    },
+    fetchOnSearch: {
+        type: Function as PropType<(search: string) => Promise<I[]>>,
+        required: false,
+        default: () => Promise.resolve([])
+    },
+    filter: {
+        type: Function as PropType<(item: I) => boolean>,
+        required: false,
+        default: () => true
+    }
+});
+
+const chipLabel = computed(() => {
+    if (model.value.length === 0) return props.label;
+    if (model.value.length === 1) {
+        const item = allItems.value.find((i) => i.id === model.value[0]);
+        return item ? item.name : "Unknown Item";
+    }
+    return `${model.value.length} selected`;
+});
+const chipColor = computed(() => {
+    return model.value.length > 0 ? "primary" : "grey lighten-1";
```

Comment:
```suggestion
    return model.value.length > 0 ? "primary" : "outline";
```
the grey etc. colors are not registered with our vuetify config, this is the same color used for the text field